app/src/main/java/io/getstream/chat/android/core/poc/library/requests/QuerySort.kt

```diff
@@ -1,6 +1,10 @@
 package io.getstream.chat.android.core.poc.library.requests
 
+import com.google.gson.annotations.JsonAdapter
+import io.getstream.chat.android.core.poc.library.json.adapters.QuerySortAdapter
 
+
+@JsonAdapter(QuerySortAdapter::class)
```

Comment:
Let's not use `@JsonAdapter`. We need to reuse `Gson` instance without passing around static instance and as far as I know `@JsonAdapter` can't help with it.
I also want to keep everything in single place: `JsonParser`, so even developers who never worked with `Gson` can see and edit everything in a single place.